lib/routes/hiringcafe/jobs.ts

```diff
@@ -0,0 +1,157 @@
+import ofetch from '@/utils/ofetch';
+import path from 'node:path';
+import { art } from '@/utils/render';
+import { Context } from 'hono';
+import { getCurrentPath } from '@/utils/helpers';
+import { Route } from '@/types';
+
+const __dirname = getCurrentPath(import.meta.url);
+
+const CONFIG = {
+    DEFAULT_PAGE_SIZE: 20,
+    MAX_PAGE_SIZE: 100,
+} as const;
+
+const API = {
+    BASE_URL: 'https://hiring.cafe/api/search-jobs',
+    HEADERS: {
+        'Content-Type': 'application/json',
+        'User-Agent': 'RSSHub/1.0',
+    },
+} as const;
+
+interface GeoLocation {
+    readonly lat: number;
+    readonly lon: number;
+}
+
+interface JobInformation {
+    readonly title: string;
+    readonly description: string;
+}
+
+interface ProcessedJobData {
+    readonly company_name: string;
+    readonly is_compensation_transparent: boolean;
+    readonly yearly_min_compensation?: number;
+    readonly yearly_max_compensation?: number;
+    readonly workplace_type?: string;
+    readonly requirements_summary?: string;
+    readonly job_category: string;
+    readonly role_activities: readonly string[];
+    readonly formatted_workplace_location?: string;
+}
+
+interface JobResult {
+    readonly id: string;
+    readonly apply_url: string;
+    readonly job_information: JobInformation;
+    readonly v5_processed_job_data: ProcessedJobData;
+    readonly _geoloc: readonly GeoLocation[];
+    readonly estimated_publish_date: string;
+}
+
+interface ApiResponse {
+    readonly results: readonly JobResult[];
+    readonly total: number;
+}
+
+interface SearchParams {
+    readonly keywords: string;
+    readonly page?: number;
+    readonly size?: number;
+}
+
+const validateSearchParams = ({ keywords, page = 0, size = CONFIG.DEFAULT_PAGE_SIZE }: SearchParams): SearchParams => {
+    if (!keywords?.trim()) {
+        throw new Error('Keywords cannot be empty');
+    }
+
```

Comment:
```suggestion
```
Unnecessary check since https://github.com/DIYgod/RSSHub/blob/90973a4f21548db6f6a7b4ca06eb5128b56d4f8f/lib/routes/hiringcafe/jobs.ts#L137 defines `keywords` as compulsory parameter. Requests leaving `keywords` empty will be rejected by middleware.